spark/src/main/scala/org/apache/spark/sql/delta/commands/OptimizeTableCommand.scala

```diff
@@ -273,34 +280,17 @@ class OptimizeExecutor(
 
       val maxThreads =
         sparkSession.sessionState.conf.getConf(DeltaSQLConf.DELTA_OPTIMIZE_MAX_THREADS)
-      val updates = ThreadUtils.parmap(jobs, "OptimizeJob", maxThreads) { partitionBinGroup =>
-        runOptimizeBinJob(txn, partitionBinGroup._1, partitionBinGroup._2, maxFileSize)
-      }.flatten
-
-      val addedFiles = updates.collect { case a: AddFile => a }
-      val removedFiles = updates.collect { case r: RemoveFile => r }
-      val removedDVs = filesToProcess.filter(_.deletionVector != null).map(_.deletionVector).toSeq
-      if (addedFiles.size > 0) {
-        val metrics = createMetrics(sparkSession.sparkContext, addedFiles, removedFiles, removedDVs)
-        commitAndRetry(txn, getOperation(), updates, metrics) { newTxn =>
-          val newPartitionSchema = newTxn.metadata.partitionSchema
-          val candidateSetOld = candidateFiles.map(_.path).toSet
-          val candidateSetNew = newTxn.filterFiles(partitionPredicate).map(_.path).toSet
-
-          // As long as all of the files that we compacted are still part of the table,
-          // and the partitioning has not changed it is valid to continue to try
-          // and commit this checkpoint.
-          if (candidateSetOld.subsetOf(candidateSetNew) && partitionSchema == newPartitionSchema) {
-            true
-          } else {
-            val deleted = candidateSetOld -- candidateSetNew
-            logWarning(s"The following compacted files were delete " +
-              s"during checkpoint ${deleted.mkString(",")}. Aborting the compaction.")
-            false
-          }
-        }
+      val batchResults = batchSize match {
+        case Some(size) =>
+          groupBinsIntoBatches(jobs, size).map(runOptimizeBatch(_, maxFileSize))
```

Comment:
Hmm didn't look at the ConflictChecker when working on this, I did have in my mind an improvement to doing multiple batches simultaneously/overlapping to prevent the tail of execution for each batch. There's no conceptual reason the commits should conflict, since they are reading specific files and not changing data. If they would conflict that might just be an improvement that should be made in the ConflictChecker

Reply:
Though since I'm creating a transaction off the same original snapshot, even doing them serially they should invoke the same conflict checking against themselves right? And the current simple test doesn't seem to have an issue. Will see if I encounter anything in further tests

Reply:
Actually I guess the Optimize command does it's own conflict checking to resolve that

Reply:
Yes, there isn't conflict within a OPITMIZE command since commits from different batches are working on disjoint set of files as you mentioned.  It is more of the conflicts between OPTIMIZE commands.  Say if batch1 from OPTIMIZE A runs concurrently with batch1 from OPTIMIZE B, and the batch mixes multiple partition files, those two batches are likely conflicting since there is high chance files are overlapped in some partition. Had if batches are at partition boundaries,  conflict would happen less likely.  This is due to Delta enhanced conflict resolution for partitioned table.
This won't eliminate conflict between OPTIMIZEs, but more of a optimization. I don't see it is needed now and we can reivist if there is one.